src/TTrigger.cpp

```diff
@@ -419,155 +432,158 @@ bool TTrigger::match_perl(char* subject, const QString& toMatch, int regexNumber
+                int begin = *iti;
+                std::string& s = *its;
+                int length = QString::fromStdString(s).size();
+                if (total > 1) {
+                    // skip complete match in Perl /g option type of triggers
+                    // to enable people to highlight capture groups if there are any
+                    // otherwise highlight complete expression match
+                    if (i % numberOfCaptureGroups != 1) {
+                        pC->selectSection(begin, length);
+    #if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+                        if (mBgColor != QColorConstants::Transparent) {
+    #else
+                        if (mBgColor != QColor("transparent")) {
+    #endif
+                            pC->setBgColor(r1, g1, b1, 255);
+                        }
+    #if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+                        if (mFgColor != QColorConstants::Transparent) {
+    #else
+                        if (mFgColor != QColor("transparent")) {
+    #endif
+                            pC->setFgColor(r2, g2, b2);
+                        }
+                    }
+                } else {
+                    pC->selectSection(begin, length);
+                    pC->setBgColor(r1, g1, b1, 255);
+                    pC->setFgColor(r2, g2, b2);
+                }
+            }
+            pC->reset();
+        }
+        if (mIsMultiline) {
+            updateMultistates(regexNumber, captureList, posList, &nameGroups);
+            return;
+        } else {
+            TLuaInterpreter* pL = mpHost->getLuaInterpreter();
+            pL->setCaptureGroups(captureList, posList);
+            pL->setCaptureNameGroups(nameGroups, namePositions);
+            execute();
+            pL->clearCaptureGroups();
+            if (mFilterTrigger) {
+                if (captureList.size() > 1) {
+                    int total = captureList.size();
+                    auto its = captureList.begin();
+                    auto iti = posList.begin();
+                    for (int i = 1; iti != posList.end(); ++iti, ++its, i++) {
+                        int begin = *iti;
+                        std::string& s = *its;
+                        if (total > 1) {
+                            // skip complete match in Perl /g option type of triggers
+                            // to enable people to highlight capture groups if there are any
+                            // otherwise highlight complete expression match
+                            if (i % numberOfCaptureGroups != 1) {
+                                filter(s, begin);
+                            }
+                        } else {
+                            filter(s, begin);
+                        }
+                    }
+                }
+            }
+            return;
+        }
+    }
+}
+
+bool TTrigger::match_begin_of_line_substring(const QString& toMatch, const QString& regex, int regexNumber, int posOffset)
```

Comment:
:thinking: Given that many of these methods do NOT involve an actual PCRE would it be clearer to replace the argument names with the *traditional* "needle" and "haystack" terms...? :grinning:

Reply:
Hmm, I had no idea they're the traditional terms?

Reply:
Well,  for example see this paragraph: https://sodocumentation.net/julia-lang/topic/5890/regexes#finding-matches, some of the Qt 6 classes use it, at least: **[QStringTokenizer](https://doc.qt.io/qt-6/qstringtokenizer.html)** and **[QStaticByteArrayMatcher](https://doc.qt.io/qt-6.2/qstaticbytearraymatcher.html)**.

Reply:
Fair enough.

Reply:
Fixed up.